bdikit/mapping_algorithms/scope_reducing/_algorithms/contrastive_learning/cl_api.py

```diff
@@ -105,15 +114,58 @@ def _sample_to_15_rows(self, table: pd.DataFrame):
                 table = unique_rows.sample(n=15, random_state=1)
         return table
 
+    def _check_gdc_cache(self, table: pd.DataFrame):
+
+        gdc_df = pd.read_csv(GDC_TABLE_PATH)
+        gdc_hash = hash_dataframe(gdc_df)
+
+        table_hash = hash_dataframe(table)
+
+        df_hash_file = None
+        features = None
+
+        # check if table for computing embedding is the same as the GDC table we have in resources
+        if table_hash == gdc_hash:
+            df_hash_file = os.path.join(BDIKIT_EMBEDDINGS_CACHE_DIR, gdc_hash)
+            # Found file in cache
+            if os.path.isfile(df_hash_file):
+                try:
+                    # Load embeddings from disk
+                    with open(df_hash_file, "r") as file:
+                        features = [
+                            [float(val) for val in vec.split(",")]
+                            for vec in file.read().split("\n")
+                            if vec.strip()
+                        ]
+                        if len(features) != len(gdc_df.columns):
+                            features = None
+                            raise ValueError("Mismatch in the number of features")
+                except Exception as e:
+                    print(f"Error loading features from cache: {e}")
+                    features = None
+        return df_hash_file, features
+
     def _load_table_tokens(self, table: pd.DataFrame) -> List[np.ndarray]:
+
+        embedding_file, embeddings = self._check_gdc_cache(table)
```

Comment:
The current implementation will read the whole GDC_TABLE_PATH file every time we call this function. So it would be nice if `_check_gdc_cache` could avoid reading that file every time (see the other comment above).

Reply:
Yes. See comment below.